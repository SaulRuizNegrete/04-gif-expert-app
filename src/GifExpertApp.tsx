import { useState } from 'react'

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch', 'Dragon Ball'])

    const onAddCategory = () => {
        //setCategories([...categories, 'Valorant'])
        setCategories( cat => [...cat, 'Valorant'])
    }

  return (
    <>
        {/** Titulo */}
        <h1>GifExpertApp</h1>

        {/**Input */}

        {/**Listado de Gif */}
        <button onClick={ onAddCategory }>Agregar</button>

        <ol>
            { 
                categories.map( category => {
                    return <li key={ category }>{ category }</li>
                }) 
            }
        </ol>

        {/**Gif Item */}

    </>
  )
}
