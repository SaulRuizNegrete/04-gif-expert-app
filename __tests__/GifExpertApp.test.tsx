import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import { GifExpertApp } from '../src/GifExpertApp'

describe('pruebas en GifExpertApp Component', () => { 

    const newCategory = 'New Category'

    test('debe hacer match con el snapshot ', () => { 
        const { container } = render(<GifExpertApp />) 
        expect(container).toMatchSnapshot()
     })

     test('debe de agregar una nueva categorÃ­a', () => { 
        
        render(<GifExpertApp />)

        const input = screen.getByRole('textbox') as HTMLInputElement
        const form = screen.getByRole('form')

        fireEvent.input(input, { target: { value: newCategory } })
        fireEvent.submit(form)

        expect(screen.getAllByRole('heading', { level: 3 }).length).toBe(2);

      })
     
     test('No debe agregar una categoria repetida', () => { 
        
        render(<GifExpertApp />)

        const input = screen.getByRole('textbox') as HTMLInputElement
        const form = screen.getByRole('form')

        fireEvent.input(input, { target: { value: '' } })
        fireEvent.submit(form)

        expect(screen.getAllByRole('heading', { level: 3 }).length).toBe(1);
        
      })
 })