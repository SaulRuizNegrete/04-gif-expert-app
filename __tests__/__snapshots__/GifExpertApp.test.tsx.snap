// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifExpertApp Component debe hacer match con el snapshot  1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <button
    aria-label="btn-add"
  >
    Agregar
  </button>
  <h3>
    One Punch
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
